### PR TITLE
Foster Application Page

### LINKED ISSUE #34: Foster Application Page

### PR DESCRIPTION
**Pull request by petabite**
### Administrative Info
[GitHub Issue #34]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
- added ApplicationProgress and various Form components
- created Foster Application and Foster Agreement page (`/application`)
- fixed scrolling issue noted in #20 

### Testing
- tested form validation works
- tested view changes appropriately on mobile breakpoint

### Confirmation of Change 
image

### CODE REVIEW

File: client/src/pages/Application.js [hunk 1]
```diff
@@ -1,10 +1,506 @@
...
+  );
+
+  const onSubmit = (data) => {
+    console.log(data);
+    setView("agreement");
+  };
+
+  const onError = (errors) => {
+    console.log(errors);
+    // setView("agreement"); // uncomment this to see the foster agreement w/o filling out the form
```
Comment (petabite) [inline client/src/pages/Application.js:54]: You can use this to see the foster agreement

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Looks clean Philip! Here's some things I noticed though:
1. I think you can remove the increment/decrement buttons for the zipcode field. They're not too useful there.
2. There seems to be a bit of scroll on the overall page. Can you get rid of this and have the form section extend all the way down to the bottom of the screen? Same with the form agreement section. Should look like this:
application_page

3. When the user inputs "n/a" for landlord, make the landlord phone number and email fields optional. Or you can allow "n/a" inputs for those as well. I think the former would be ideal though.
4. Can you add an "Exit" button in the top left corner similar to the design? This should just redirect to the dashboard page.
5. I think you can also remove the scrollbar within the form section since we have the trackbar on the left side. Let me know what you think about this though.

**Review by willji0023 — CHANGES_REQUESTED** [review]
Everything looks really great! Here's just some minor things I noticed:

1. I think the margins around the milestone text "Application Submitted" on Mobile view might be a little squished and could be increased a tad on either side, if possible. 
image

2. It would also be nice if the icon of the calender button on the date field for the Foster Agreement form matched the one on the Figma.

3. Not sure if we plan on using some sort of signature box for the Foster Agreement form, but if so maybe for now we could at least increase the size of the text-box like in the Figma? Maybe @dnanjunda can fact-check me on this.
image

**Comment by dnanjunda** [comment]
> 
Yes, we'll leave the signature box as is for now until I talk to the client about it.

**Comment by petabite** [comment]
> 2. It would also be nice if the icon of the calender button on the date field for the Foster Agreement form matched the one on the Figma.

Yeah, the icon you see comes from the browser's default date input implementation. Would it be okay if we left it as the default, @dnanjunda?

**Review by willji0023 — APPROVED** [review]
Everything looks really clean! And looks like all of both Dhanush's and my comments were addressed. LGTM

**Review by dnanjunda — APPROVED** [review]
LGTM! Great work Philip

**Pull request merged**